### PR TITLE
Ls 17 to MAIN

### PR DESCRIPTION
**Pull request by special-ded**
Ls 17 to MAIN

### CODE REVIEW

**Review by philippbulanin — APPROVED** [review]

File: store/src/app/administration/administration-routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    path: '',
+    redirectTo: 'users',
+    pathMatch: 'full'
+  },
+  {
+    path: '',
+    component: AdministrationComponent,
+    children: [
+      {
+        path: 'login',
```
[the remark this answers is not in the document]
Comment (special-ded) [reply store/src/app/administration/administration-routing.module.ts:20]: understood

File: store/src/app/administration/sidebar/sidebar.component.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-sidebar',
+  templateUrl: './sidebar.component.html',
+  styleUrls: ['./sidebar.component.scss']
+})
+export class SidebarComponent {
+
+  @Input() active: boolean = false
```
[the remark this answers is not in the document]
Comment (special-ded) [reply store/src/app/administration/sidebar/sidebar.component.ts:10]: done

File: store/src/app/administration/users/users.component.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<p>users works!sssssssssssssssssssssssssssssddddddddddddddddddddddddddddddddddddddddddddddddd</p>
```
[the remark this answers is not in the document]
Comment (special-ded) [reply store/src/app/administration/users/users.component.html:1]: done

File: store/src/app/administration/table/table.component.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  selector: 'app-table',
+  templateUrl: './table.component.html',
+  styleUrls: ['./table.component.scss']
+})
+
+export class TableComponent implements OnInit {
+
+  constructor(private productsService: ProductsService) { }
+
+  products: Product[] = [];
```
[the remark this answers is not in the document]
Comment (special-ded) [reply store/src/app/administration/table/table.component.ts:11]: done

File: store/src/app/products/services/cart.service.ts [hunk 5]
```diff
@@ -11,55 +12,63 @@ export class CartService implements OnInit {
...
-    this.cartList = [...this.cartList, product]
+    this.cartList.push(product);
     this.lsService.setToLS(this.cartList);
-    this.getTotalPrice();
+    this.updateTotalPrice();
+
     return this.cartList;
   }
 
   removeFromCart(id: number) {
```
[the remark this answers is not in the document]
Comment (special-ded) [reply store/src/app/products/services/cart.service.ts:32]: done

**Pull request merged**